### PR TITLE
Enabled CBO by default

### PR DESCRIPTION
**Pull request by arhimondr**

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java [hunk 1]
```diff
@@ -114,12 +114,11 @@ public void testDefaults()
                 .setCreatesOfNonManagedTablesEnabled(true)
                 .setHdfsWireEncryptionEnabled(false)
                 .setPartitionStatisticsSampleSize(100)
-                .setIgnoreCorruptedStatistics(false)
+                .setIgnoreCorruptedStatistics(true)
                 .setRecordingPath(null)
                 .setRecordingDuration(new Duration(0, TimeUnit.MINUTES))
                 .setReplay(false)
-                .setCollectColumnStatisticsOnWrite(false)
```
Comment (picrinite) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java]: Duplication is fixed now

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java [hunk 2]
```diff
@@ -296,12 +295,11 @@ public void testExplicitPropertyMappings()
                 .setCreatesOfNonManagedTablesEnabled(false)
                 .setHdfsWireEncryptionEnabled(true)
                 .setPartitionStatisticsSampleSize(1234)
-                .setIgnoreCorruptedStatistics(true)
+                .setIgnoreCorruptedStatistics(false)
                 .setRecordingPath("/foo/bar")
                 .setRecordingDuration(new Duration(42, TimeUnit.SECONDS))
                 .setReplay(true)
-                .setCollectColumnStatisticsOnWrite(true)
-                .setCollectColumnStatisticsOnWrite(true)
```
Comment (picrinite) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java]: Ditto

**Comment by rschlussel** [comment]
Travis failures are related.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java (hunk 1 above)
Comment (arhimondr) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java]: Extracted into a separate commit

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java (hunk 2 above)
Comment (arhimondr) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveClientConfig.java]: Extracted into a separate commit

**Comment by tooptoop4** [comment]
bump

**Comment by arhimondr** [comment]
@tooptoop4 There were concerns that our documentation might not be good enough to have this feature enabled by default. Currently i don't have extra cycles to be able to improve the documentation. I'm not exactly sure if we want to merge this feature before we do so. @rschlussel, what do you think?

**Pull request closed without merging**